gramps_webapi/__main__.py

```diff
@@ -58,12 +59,71 @@ def cli(ctx, config):
 
 @cli.command("run")
 @click.option("-p", "--port", help="Port to use (default: 5000)", default=5000)
-@click.option("--tree", help="Tree ID", default=None)
+@click.option("-t", "--tree", help="Tree ID: '*' for multi-trees", default=None)
+@click.option(
+    "-o",
+    "--open",
+    help="Open gramps-web in browser: 'tab', 'window', or 'no'",
+    default="no",
+)
+@click.option(
+    "-d",
+    "--debug-level",
+    help="Debug level: 'info', 'debug', 'warning', 'critical'",
+    default="info",
+)
+@click.option("-l", "--log-file", help="Set logging file to this path", default=None)
+@click.option(
+    "--host", help="Set the host address for server to listen on", default="127.0.0.1"
+)
+@click.option(
+    "--max-workers",
+    help="Maximum number of workers for frontend; requires --use-wsgi",
+    default=None,
+)
+@click.option("--use-wsgi", is_flag=True, help="Add a wsgi wrapper around server")
 @click.pass_context
-def run(ctx, port, tree):
+def run(ctx, port, tree, host, open, debug_level, log_file, max_workers, use_wsgi):
     """Run the app."""
     app = ctx.obj["app"]
-    app.run(port=port, threaded=True)
+    debug_level = debug_level.upper()
+    if max_workers is None:
+        max_workers = min(32, os.cpu_count() + 4)
+
+    def open_webbrowser():
+        time.sleep(1.0)
+        new = {"tab": 0, "window": 1}[open]
```

Comment:
This will raise a `KeyError` if the user provides any unexpected string. I think it would be better to use a [click choice option](https://click.palletsprojects.com/en/stable/options/#choice-options) for the browser flag.

Reply:
I'll take a look.

Reply:
Did you forget to commit those changes? I don't see them in the final PR.

Reply:
I used the [click.Choice](https://github.com/gramps-project/gramps-web-api/pull/576/files#diff-fdcef86582d5df066774fb9feb6f20ca30750f97d63064aebec9e4246c7d0e3cR69), but I didn't want to make the choices be "1" or "2" but "window" or "tab", so still need the dict. But the value is guaranteed to be in dict.

Reply:
Must have overlooked it, thanks